hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -3,18 +3,1146 @@
+            register (or interrupt if available). Check the digest registers for correctness against
+            the expected digest value.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_aes_shadow_reg_alert
+      desc: '''Verify shadow reg alert from AES.
+
+            Inject a storage error via backdoor to generate this alert signal while the SW is
+            actively executing some piece of code. Verify alert propagation to an NMI.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_aes_keymgr_key
+      desc: '''Verify the keymgr interface to AES.
+
+            X-ref'ed with the keymgr test.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_aes_idle
+      desc: '''Verify AES idle signaling to clkmgr.
+
+            Details TBD.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // HMAC (pre-verified IP) integration tests:
+    {
+      name: chip_hmac_enc
+      desc: '''Verify HMAC / SHA256 operation.
+
+            SW test verifies SHA256 operation with a known key, plain text and digest (pick one of
+            the NIST vectors). SW validates the reception of hmac done and fifo empty interrupts.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_hmac_alert
+      desc: '''Verify alert from HMAC.
+
+            Details TBD.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_hmac_idle
+      desc: '''Verify HMAC idle signaling to clkmgr.
+
+            Details TBD.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // KMAC pre-verified IP) integration tests:
+    {
+      name: chip_kmac_enc
+      desc: '''Verify the SHA3 operation.
+
+            SW test verifies SHA3 operation with a known key, plain text and digest (pick one of
+            the NIST vectors). SW validates the reception of kmac done and fifo empty interrupts.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_kmac_sram_uncorrectable_alert
+      desc: '''Verify the SRAM uncorrectable alert from KMAC.
+
+            Inject 2 bit errors within the SRAM inside KMAC via backdoor and ensure that this alert
+            propagates to an NMI.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_kmac_sram_data_parity_alert
+      desc: '''Verify the data parity alert from KMAC.
+
+            Details TBD.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_kmac_keymgr_key_data
+      desc: '''Verify the keymgr interface to KMAC.
+
+            X-ref'ed with keymgr test.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: chip_kmac_idle
+      desc: '''Verify KMAC idle signaling to clkmgr.
+
+            Details TBD.
+            '''
+      milestone: V2
+      tests: []
+    }
+
+    // CSRNG tests:
+    {
```

Comment:
btw @sriyerg , this hasn't landed yet, but we should probably discuss a bit on how we want to handle the full chip RNG timing.  

The problem is that technically the RNG is supposed to go through a set of start-up checks, checks which the ROM will eventually enable and wait on.  We don't want that to incur a bunch of delay on DV, so we'll probably need to sync with various parties to see what the best approach is.

My personal preference is to make the RNG on AST super fast, so we at least test all the logic at an accelerated pace.

Reply:
Understood, thanks for the heads up. I have come across this in my past exp and I recall having conditional compile switches to speed up the DV.